### DISCUSSION TITLE
How do I upload Excel and display the table in Flet?

### DISCUSSION THREAD

**Discussion by basatsu** [flet-dev/flet#976] [Q&A]
I want to upload excel and display the data table., but I'm not sure how to implement it.
In this case, 
I guess you need to use pandas to read excel data but I'm not sure if it's impossible to do what I want in Flet.

Firstly, import flet and pandas.

```
import flet as ft
import pandas as pd
```

upload excel with FilePicker component , and read uploaded excel data like below

`myexcel=pd.read_excel("fakedata.xlsx")`

then show the data with Datatable component.

I imagine you can do like this however, how do I get the data of excel file with FilePicker?
I'd like to know how to pass excel data to Datatable like below.

```
ft.DataTable(
・・・
            columns=[
                ft.DataColumn(
                    ft.Text("Column 1"),
                    on_sort=lambda e: print(f"{e.column_index}, {e.ascending}"),
                ),
                ft.DataColumn(
                    ft.Text("Column 2"),
                    tooltip="This is a second column",
                    numeric=True,
                    on_sort=lambda e: print(f"{e.column_index}, {e.ascending}"),
                ),
            ],
            rows=[
                ft.DataRow(
                    [ft.DataCell(ft.Text("A")), ft.DataCell(ft.Text("1"))],
                    selected=True,
                    on_select_changed=lambda e: print(f"row select changed: {e.data}"),
                ),
                ft.DataRow([ft.DataCell(ft.Text("B")), ft.DataCell(ft.Text("2"))]),
            ],
  ),

```

I tried it but I couldn't figure out. Do you have any ideas?
I'd appreciate it if you could give me any sample.

**Comment by ndonkoHenri** [comment]
https://github.com/StanMathers/simple-datatable
Give this a look.

**Reply by basatsu** [reply]
Thank you.
I tried to implemeted it from file upload to display dataframe table,
but it doesn't work.  I can't display datatable after uploading file
What I want is to select file to upload then the datatable will be displayed.
Here is my code. Could you give me any advice on where I'm missing?
I'm still a Flet beginner and I'm not familiar with how to display content of uploaded file...

```
def main(page: ft.Page):
    def pick_files_result(e: ft.FilePickerResultEvent):
        selected_files.value = (
            ", ".join(map(lambda f: f.name, e.files)) if e.files else "Cancelled!"
        )
        selected_files.update()

    pick_files_dialog = ft.FilePicker(on_result=pick_files_result)
    selected_files = ft.Text()

    page.overlay.append(pick_files_dialog)

    page.add(
        ft.Row(
            [
                ft.ElevatedButton(
                    "Pick files",
                    icon=ft.icons.UPLOAD_FILE,
                    on_click=lambda _: pick_files_dialog.pick_files(
                        allow_multiple=True
                    ),
                ),
                selected_files,
            ]
        )
    )
    
    if selected_files != None:
        df = pd.read_csv(selected_files)
        simpledt_df = DataFrame(df)  # Initialize simpledt DataFrame object
        simpledt_dt = simpledt_df.datatable  # Extract DataTable instance from simpledt
        page.add(simpledt_dt)

ft.app(target=main)
```


I got 
ValueError: Invalid file path or buffer object type: <class 'flet.text.Text'>
here
`df = pd.read_csv(selected_files)`

**Reply by ttsgit** [reply]
`    if selected_files.value:
        df = pd.read_csv(selected_files.value)
        simpledt_df = DataFrame(df)  # Initialize simpledt DataFrame object
        simpledt_dt = simpledt_df.datatable  # Extract DataTable instance from simpledt
        page.add(simpledt_dt)`
try this
